src/UI/Tables/Stateable.elm

```diff
@@ -0,0 +1,746 @@
+withItems items (Table prop opt) =
+    Table prop { opt | items = items }
+
+
+
+-- State
+
+
+type Msg
+    = MobileToggle Int
+    | ForFilters Filters.Msg
+    | FilterDialogOpen Int
+    | FilterDialogClose
+
+
+type State
+    = State StateModel
+
+
+type alias StateModel =
+    { filters : Filters.Model
+    , mobileSelected : Maybe Int
+    , filterDialog : Maybe Int
+    }
+
+
+init : State
+init =
+    State { filters = Filters.init, mobileSelected = Nothing, filterDialog = Nothing }
+
+
+update : Msg -> State -> State
+update msg (State state) =
+    case msg of
+        MobileToggle index ->
+            State
+                { state
+                    | mobileSelected =
+                        if state.mobileSelected == Just index then
+                            Nothing
+
+                        else
+                            Just index
+                }
+
+        ForFilters subMsg ->
+            State { state | filters = Filters.update subMsg state.filters }
+
+        FilterDialogOpen index ->
+            State { state | filterDialog = Just index }
+
+        FilterDialogClose ->
+            State { state | filterDialog = Nothing }
+
+
+
+-- Responsive
+
+
+type alias Responsive msg item columns =
+    { toDetails : item -> Details msg columns
+    , toCover : item -> Cover
+    }
+
+
+type alias Cover =
+    ListView.ToggleableCover
+
+
+type alias Detail msg =
+    { label : String, content : Common.Cell msg }
+
+
+type alias Details msg columns =
+    NArray (Maybe (Detail msg)) columns
+
+
+withResponsive : Responsive msg item columns -> StateableTable msg item columns -> StateableTable msg item columns
+withResponsive responsive (Table prop opt) =
+    Table prop { opt | responsive = Just responsive }
+
+
+detailsEmpty : Details msg T.Zero
+detailsEmpty =
+    NArray.empty
+
+
+detailsPush : Detail msg -> Details msg columns -> Details msg (T.Increase columns)
+detailsPush detail accu =
+    NArray.push (Just detail) accu
+
+
+detailsPushHidden : Details msg columns -> Details msg (T.Increase columns)
+detailsPushHidden accu =
+    NArray.push Nothing accu
+
+
+
+-- Filters
+
+
+type alias Filters msg item columns =
+    Filters.Filters msg item columns
+
+
+type alias Strategy msgs value item =
+    Filters.Strategy msgs value item
+
+
+withFilters : Filters msg item columns -> StateableTable msg item columns -> StateableTable msg item columns
+withFilters filters (Table prop opt) =
+    Table prop { opt | filters = Just filters }
+
+
+filtersEmpty : Filters msg item T.Zero
+filtersEmpty =
+    NArray.empty
+
+
+filtersPush : Filters.Filter msg item -> Filters msg item columns -> Filters msg item (T.Increase columns)
+filtersPush filter accu =
+    NArray.push filter accu
+
+
+localSingleTextFilter :
+    Maybe String
+    -> (item -> String)
+    -> Filters msg item columns
+    -> Filters msg item (T.Increase columns)
+localSingleTextFilter initValue get accu =
```

Comment:
What do you think about moving all this logic to Filters module? I think Table should not contain the filter logic (for example, the applier), maybe we should just call a Filters function here

Reply:
That used to be impossible before the "Common" file, now I can do it! Good suggestion!